TranslationManager/Classes/Manager/TranslatableManager.swift

```diff
@@ -424,18 +432,20 @@ public class TranslatableManager<L: LanguageModel, C: LocalizationModel> {
             case .success(let translationsData):
                 // New translations downloaded
                 self.handleTranslationsResponse(translationsData: translationsData,
+                                                in: group,
                                                 completion: completion)
 
             case .failure(let error):
                 // Error downloading translations data
+                group.leave()
```

Comment:
I don't think this is a safe approach - if we are passing the group to this method synchronously, then this method should handle the entering and leaving imo. I would put this in a defer block in the completion, so that no matter the result it gets called. The handle translation method should then do the same where it also controls that it enters the group and then exits when done.

Reply:
I've changed the logic now. 

In both `updateLocaleTranslations` and `handleTranslationsResponse` this bit of logic has been added to balance out entering and leaving

```
group.enter()
        
defer {
    group.leave()
}
```